frontend/test/model/accessRequests/AccessRequestDisplay.spec.tsx

```diff
@@ -0,0 +1,77 @@
+import { ThemeProvider } from '@mui/material/styles'
+import { render, screen, waitFor } from '@testing-library/react'
+import { UserDisplayProps } from 'src/common/UserDisplay'
+import AccessRequestDisplay from 'src/model/beta/accessRequests/AccessRequestDisplay'
+import { ReviewBannerProps } from 'src/model/beta/reviews/ReviewBanner'
+import { ReviewDisplayProps } from 'src/model/beta/reviews/ReviewDisplay'
+import { lightTheme } from 'src/theme'
+import { testAccessRequest, testAccessRequestReview, testAccessRequestWithComments } from 'utils/test/testModels'
+import { describe, expect, it, vi } from 'vitest'
+
+import { useGetReviewRequestsForModel } from '../../../actions/review'
+
+vi.mock('../../../actions/review', () => ({
+  useGetReviewRequestsForModel: vi.fn(),
+}))
+vi.mock('src/model/beta/reviews/ReviewBanner.tsx', () => ({ default: (_props: ReviewBannerProps) => <></> }))
+vi.mock('src/common/UserDisplay.tsx', () => ({ default: (_props: UserDisplayProps) => <></> }))
+vi.mock('src/model/beta/reviews/ReviewDisplay.tsx', () => ({ default: (_props: ReviewDisplayProps) => <></> }))
+
+describe('AccessRequestDisplay', () => {
+  it('displays the metadata for an access request metadata object when display component is rendered', async () => {
+    vi.mocked(useGetReviewRequestsForModel).mockReturnValue({
+      reviews: [testAccessRequestReview],
+      isReviewsLoading: false,
+      isReviewsError: undefined,
+      mutateReviews: vi.fn(),
+    })
+    render(
+      <ThemeProvider theme={lightTheme}>
+        <AccessRequestDisplay accessRequest={testAccessRequest} />
+      </ThemeProvider>,
+    )
+
+    await waitFor(async () => {
+      expect(await screen.findByText(testAccessRequest.metadata.overview.name)).toBeDefined()
+    })
+  })
+
+  it('displays a comments icon when the provided access request has comment responses', async () => {
+    vi.mocked(useGetReviewRequestsForModel).mockReturnValue({
+      reviews: [testAccessRequestReview],
+      isReviewsLoading: false,
+      isReviewsError: undefined,
+      mutateReviews: vi.fn(),
+    })
+    render(
+      <ThemeProvider theme={lightTheme}>
+        <AccessRequestDisplay accessRequest={testAccessRequestWithComments} />
+      </ThemeProvider>,
+    )
+
+    await waitFor(async () => {
+      expect(await screen.findByLabelText('Comments')).toBeDefined()
```

Comment:
Can we use `ByTestId` queries? This will use the same `data-test` attributes that we use in cypress. This should be our primary way of finding elements in tests as it's much more specific. You can read more about the different queries here: https://testing-library.com/docs/react-testing-library/cheatsheet/ and here: https://testing-library.com/docs/queries/about/#priority

As an example, this test would still pass if we happened to put the text "Comments" anywhere else in this component, even if we removed the icon. So the test isn't providing us the security it was written for. If we use `findByTestId` then it will only pass so long as that specific icon is present.

In this case we could use `findByTestId`, which returns the found element if found and throws and error when not found.

I won't flag all of these up, but if you agree, we should apply this to all of the unit tests in this PR.

Reply:
I've changed this where I can, there's a few places where I've left by "getByRole" like for the  loading spinner because the element will technically always be there, but when the LoadingButton is in a loading state it only has its role changed.

I'm also going to try my best to update the form fields, we _might_ still need to do text checking in the document. In the past I've managed to pass data-test IDs into the custom form field components as props but they don't always get attached to the correct element because of how MUI works.